connect-inject/endpoints_controller.go

```diff
@@ -334,12 +340,49 @@ func (r *EndpointsController) createServiceRegistrations(pod corev1.Pod, service
 				AliasService: serviceID,
 			},
 		},
-		Connect: nil,
 	}
 	if len(tags) > 0 {
 		proxyService.Tags = tags
 	}
 
+	tproxyEnabled, err := transparentProxyEnabled(pod, r.EnableTransparentProxy)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if tproxyEnabled {
+		var k8sService corev1.Service
+
+		err := r.Client.Get(r.Context, types.NamespacedName{Name: serviceEndpoints.Name, Namespace: serviceEndpoints.Namespace}, &k8sService)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// Check if the service has a valid IP.
+		parsedIP := net.ParseIP(k8sService.Spec.ClusterIP)
+		if parsedIP != nil {
+			taggedAddresses := make(map[string]api.ServiceAddress)
+			for _, servicePort := range k8sService.Spec.Ports {
+				taggedAddressKey := clusterIPTaggedAddressName
+				if servicePort.Name != "" {
+					taggedAddressKey = fmt.Sprintf("%s-%s", clusterIPTaggedAddressName, servicePort.Name)
+				}
+
+				taggedAddresses[taggedAddressKey] = api.ServiceAddress{
+					Address: k8sService.Spec.ClusterIP,
+					Port:    int(servicePort.Port),
+				}
+			}
+
+			service.TaggedAddresses = taggedAddresses
+			proxyService.TaggedAddresses = taggedAddresses
+
+			proxyService.Proxy.Mode = api.ProxyModeTransparent
+		} else {
+			r.Log.Info("skipping syncing service cluster IP to Consul", "name", k8sService.Name, "ns", k8sService.Namespace, "ip", k8sService.Spec.ClusterIP)
```

Comment:
should this be debug level? Wondering if there's tons of services the you don't want this. Although I guess this only applies to headless services?

Reply:
Hmm, yeah this applies only to headless services. But maybe still this should be `debug` because it's not really useful to you since everything will still work just using pod IPs. I'll change it to debug.

Reply:
or I won't 😄 It looks like zap logger doesn't have `debug` log level?